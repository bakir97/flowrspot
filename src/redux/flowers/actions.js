import axios from "axios";
import { devKeys } from "../../globals/keys";
import { LOADING_FLOWERS, SUCCESS_FLOWERS } from "../constants";
export const getFlowers = () => async dispatch => {
  try {
    dispatch(loadingFlowers(true));
    const flowers = await axios.get(`${devKeys.api}/flowers`, {
      headers: { Authorization: devKeys.getFlowersAuth }
    });
    dispatch(successFlowers(flowers.data.flowers));
    dispatch(loadingFlowers(false));
  } catch (error) {
    dispatch(loadingFlowers(false));
  }
};
export const searchFlowers = search => async dispatch => {
  try {
    dispatch(loadingFlowers(true));

    const flowers = await axios.get(
      `${devKeys.api}/flowers/search?query=${search}`,
      {
        headers: { Authorization: devKeys.flowersSearchAuth }
      }
    );
    dispatch(successFlowers(flowers.data.flowers));
    dispatch(loadingFlowers(false));
  } catch (error) {
    console.log(error);

    dispatch(loadingFlowers(false));
  }
};
export const loadingFlowers = payload => ({
  type: LOADING_FLOWERS,
  payload
});
export const successFlowers = payload => ({
  type: SUCCESS_FLOWERS,
  payload
});
