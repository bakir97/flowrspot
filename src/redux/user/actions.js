import axios from "axios";
import { devKeys } from "../../globals/keys";
import { SUCCESS_USER, LOADING_USER, ERROR_USER, SET_USER } from "../constants";
import { getFlowers } from "../flowers/actions";
export const registerUser = data => async dispatch => {
  try {
    dispatch(loadingUser(true));
    await axios.post(`${devKeys.api}/users/register`, data);
    dispatch(successUser(true));
    dispatch(loadingUser(false));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser(error.response.data.error));
  }
};
export const loginUser = data => async dispatch => {
  try {
    dispatch(loadingUser(true));
    const user = await axios.post(`${devKeys.api}/users/login`, data);
    dispatch(successUser(true));

    const userData = {
      auth_token: user.data.auth_token,
      email: data.email
    };
    dispatch(login(userData));
    dispatch(getFlowers(user.data.auth_token));
    localStorage.setItem("user", JSON.stringify(userData));
    dispatch(loadingUser(false));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser([error.response.data.error]));
  }
};
export const login = user => async dispatch => {
  try {
    const userCreds = await axios.get(`${devKeys.api}/users/me`, {
      headers: { Authorization: user.auth_token }
    });
    dispatch(setUser({ ...userCreds.data.user, email: user.email }));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser([error.response.data.error]));
  }
};
export const logout = () => dispatch => {
  localStorage.removeItem("user");
  dispatch(getFlowers());
  dispatch(setUser(null));
};
export const successUser = payload => ({
  type: SUCCESS_USER,
  payload
});
export const setUser = payload => ({
  type: SET_USER,
  payload
});
export const loadingUser = payload => ({
  type: LOADING_USER,
  payload
});
export const errorUser = payload => ({
  type: ERROR_USER,
  payload
});
