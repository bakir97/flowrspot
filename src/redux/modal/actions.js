import { OPEN_MODAL, CLOSE_MODAL } from "../constants";
import { successUser } from "../user/actions";
export const openModal = type => dispatch => {
  dispatch(openModalAction(type));
  dispatch(successUser(false));
};
export const openModalAction = payload => ({
  type: OPEN_MODAL,
  payload
});

export const closeModal = () => async dispatch => {
  dispatch(closeModalAction());
  dispatch(successUser(false));
};
export const closeModalAction = () => ({
  type: CLOSE_MODAL,
  payload: null
});
