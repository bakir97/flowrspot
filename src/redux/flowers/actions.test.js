import configureMockStore from "redux-mock-store";
import { successFlowers, loadingFlowers, addNewFavFlower } from "./actions";
import thunk from "redux-thunk";
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("action creators flowers", () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  it("should set sucess action with payload true", () => {
    store.dispatch(successFlowers([{ id: 1 }, { id: 1 }, { id: 1 }]));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should set loading action with payload true", () => {
    store.dispatch(loadingFlowers(true));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should set addNewFavFlower action with payload flower id", () => {
    store.dispatch(addNewFavFlower(5));
    expect(store.getActions()).toMatchSnapshot();
  });
});
