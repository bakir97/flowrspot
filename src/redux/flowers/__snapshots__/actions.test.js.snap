// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators flowers should set addNewFavFlower action with payload flower id 1`] = `
Array [
  Object {
    "payload": 5,
    "type": "ADD_FAV",
  },
]
`;

exports[`action creators flowers should set loading action with payload true 1`] = `
Array [
  Object {
    "payload": true,
    "type": "LOADING_FLOWERS",
  },
]
`;

exports[`action creators flowers should set sucess action with payload true 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "id": 1,
      },
      Object {
        "id": 1,
      },
      Object {
        "id": 1,
      },
    ],
    "type": "SUCCESS_FLOWERS",
  },
]
`;
